tests/test_kernel_density_estimation_classifier.py

```diff
@@ -0,0 +1,127 @@
+"""Unit tests for the KNN classifier."""
+
+import math
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.learning.kernel_density_estimation_classifier \
+    import KernelDensityEstimationClassifier
+from geomstats.learning.radial_kernel_functions import triangular_radial_kernel
+
+TOLERANCE = 1e-4
+
+
+class TestKernelDensityEstimationClassifier(geomstats.tests.TestCase):
+    """Class defining the Kernel Density Estimation Classifier tests."""
+
+    def setUp(self):
+        """Define the parameters to test."""
+        gs.random.seed(1234)
+        self.radius = math.inf
+        self.dim = 1
+        self.space = Euclidean(dim=self.dim)
+        self.distance = self.space.metric.dist
+
+    @geomstats.tests.np_only
+    def test_predict(self):
+        """Test the 'predict' class method."""
+        training_dataset = gs.array([[0], [1], [2], [3]])
+        labels = [0, 0, 1, 1]
+
+        kde = KernelDensityEstimationClassifier(
+            radius=self.radius,
+            distance=self.distance)
+        kde.fit(training_dataset, labels)
+        result = kde.predict([[1.1]])
```

Comment:
why the double [[ ]] here? We have moved away from the convention that encapsulated every single vector in a [[1, 2, 3]] type array. Would only [1.1] work?

Reply:
For now it looks like it does not work if I just remove one of the two brackets.

Reply:
Could be that the vectorization does not work well for 1D vectors, because it mixes them up with scalars. Did you try on 2D vectors?